packages/gatsby-source-contentful/src/gatsby-node.js

```diff
@@ -58,11 +59,19 @@ exports.sourceNodes = async (
     return
   }
 
+  options = { ...options, ...defaultOptions }
```

Comment:
wooops I would break custom `host` and `environment` here and override it everytime with defaults